include/gc/Dialect/CPURuntime/IR/CPURuntimeOps.td

```diff
@@ -29,5 +28,91 @@ def CPURuntime_PrintfOp : CPURuntime_Op<"printf", [MemoryEffects<[MemWrite]>]>,
   }];
 }
 
+def CPURuntime_AllocOp : CPURuntime_Op<"alloc", [
+    AttrSizedOperandSegments,
+    MemoryEffects<[MemAlloc<DefaultResource>]>
+  ]> {
+
+  let summary = "CPURuntime memory allocation operation in a stack way.";
+  let description = [{
+    The `cpuruntime.alloc` operation allocates a region of memory on the CPU. It is
+    similar to the `std.alloc` op.
+
+    Example:
+
+    ```mlir
+    // For dynamic size memref
+    %memref = cpuruntime.alloc (%width) : memref<64x?xf32>
+    
+    // For static size memref
+    %memref = cpuruntime.alloc () : memref<64x32xf32>
```

Comment:
Just a discussion: 1) `alloc` or `main_alloc`? 2) shall we merge `alloc` with `thread_alloc`, and add an attr like `cpuruntime.alloc () {main} : memref<64x32xf32>`. For Reference, https://github.com/intel/mlir-extensions/blob/417a44959726f38b36ba494ff25e18c331c956bb/include/imex/Dialect/GPUX/IR/GPUXOps.td#L192

They have a `shared` attr for `gpux.alloc`.

I have no preference on merging `alloc`/`thread_alloc` or splitting them to two ops.

Reply:
I think we can merge `alloc` and `thread_alloc` to simplify the op list. Maybe I can do this refactor in the next PR, as we may need to add more allocators.